Add libdef for detox

coderjonny opened the pull request:
Library definition for Detox. 🙏🏽
https://github.com/wix/detox

related issue: https://github.com/wix/detox/issues/275

rotemmiz commented on definitions/npm/detox_v7.0.x/test_detox_v7.0.x.js:
```diff
@@ -0,0 +1,170 @@
+expect(element(by.id('UniqueId204'))).toNotExist()
+
+// waitFor API
+waitFor(element(by.id('UniqueId204'))).toBeNotVisible()
```
waitFor only works with `withTimeout(Number)`, the below lines will never trigger anything. is there a way to enforce it with flow ?

skovhus replied:
I cannot really see anything we can do here using Flow.

rotemmiz commented on definitions/npm/detox_v7.0.x/test_detox_v7.0.x.js:
```diff
@@ -0,0 +1,170 @@
+
+// Actions on element
+// ------------------
+ element(by.id('testElement')).tap()
```
The whole API is async (promise based), can flow enforce async usage across the API ?

skovhus replied:
We cannot enforce the using to `away` on these. 

But I think all the `=> {}` and `=> void` should be replace by `=> Promise<void>`.

coderjonny replied:
changed return types to `Promise<void>`. 👍

rotemmiz commented on definitions/npm/detox_v7.0.x/test_detox_v7.0.x.js:
```diff
@@ -0,0 +1,170 @@
+const welcome_text = "Shake your phone to open the developer menu.";
+expect(element(by.text(welcome_text))).toBeVisible();
+const btn = element(by.id(e2e.NATIVE_BUTTON));
+btn.tap();
```
Seems like everything is here. I have a few questions though, written above.

Thanks for taking care of Detox :)

skovhus commented:
Really nice work here. 

I do suggest that all async function should return `Promise<void>` instead of `void` and `{}`.

Let me know if you need any help landing this. : )

AndrewSouthpaw commented:
I'll wait on that to happen before we merge then. :)

adamterlson commented:
Any motion here?

AndrewSouthpaw commented:
Just waiting on someone to do the work suggested by @skovhus.

coderjonny commented:
Hey guys, 

Sorry for the delay .. I've updated name of the folder to test `detox 7.x.x` and updated some more flowTypes. Let me know if that works 👍
